plugins/module_utils/foreman_helper.py

```diff
@@ -202,6 +202,32 @@ def find_resources_by_name(self, resource, names, **kwargs):
     def find_resources_by_title(self, resource, titles, **kwargs):
         return [self.find_resource_by_title(resource, title, **kwargs) for title in titles]
 
+    def find_operatingsystem(self, name, params=None, failsafe=False, thin=None):
+        search = 'title~"{}"'.format(name)
+        list_params = {}
+        if params is not None:
+            list_params.update(params)
+        if thin is None:
+            thin = self._thin_default
+        list_params['thin'] = thin
+        results = self.list_resource('operatingsystems', search, list_params)
+        if len(results) >= 1:
+            # return the best match (with the shortest title)
+            result = sorted(results, key=lambda x: len(x['title']))[0]
```

Comment:
What happens if the user enters `CentOS 7`, but they only have `CentOS 7.3.123` and `CentOS 7.4.123` defined? Given both entries have the same length, the result is (potentially) unstable?

Reply:
yes, sadly. It's the same Problem if you had `CentOS 1.2.345` without a desription and another one with `CentOS unstable` as a description.

Reply:
With a little more effort, we can error out on such ambiguities.

Reply:
Just my opinion, if we have multiple results for the fuzzy search and among them is one perfect match, return that, otherwise fail with 'search result not distinct' or something like that.
Because, If you try to match the shortest result and there are multiple results, it's always a coin-flip.

Reply:
@m-bucher do I read this as a "yay" to my "refering operating systems" comment in https://github.com/theforeman/foreman-ansible-modules/issues/373#issuecomment-516737370?

Reply:
That corresponds to the second part of that answer, which i agree:
https://github.com/theforeman/foreman-ansible-modules/issues/373#issuecomment-516737370
I think, we don't have to call the api twice though.

Reply:
Yeah, how you *implement* that is a detail. Doing two API calls would have the benefit that you can just use `find_resource` twice instead of effectively duplicating its content.